code/game/machinery/metaldetector.dm

```diff
@@ -0,0 +1,69 @@
+/obj/machinery/metal_detector
+	name = "metal detector"
+	desc = "A advanced metal detector used to detect weapons."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "metal_detector"
+	anchored = 1
+	req_one_access = list(access_security, access_judge, access_heads, access_bodyguard)
+	var/list/banned_objects=list(/obj/item/weapon/gun,
+								/obj/item/weapon/material,
+								/obj/item/weapon/melee,
+								/obj/item/device/transfer_valve,
+								/obj/item/weapon/grenade/,
+								/obj/item/ammo_casing/,
+								/obj/item/ammo_magazine
+								)
+	plane = MOB_PLANE
+	layer = ABOVE_MOB_LAYER
+
+/obj/machinery/metal_detector/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/metal_detector(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/console_screen(src)
+	RefreshParts()
+
+/obj/machinery/metal_detector/attackby(obj/item/W, mob/usr)
```

Comment:
`usr` should be `user` in this entire proc

Reply:
Also you don't call `..()` at the end of attackby()

Reply:
@ike709 What is the difference between usr and user if they both work? Also, nothing is calling . . ( ) at the end of attackby?